c++/src/Reader.cc

```diff
@@ -764,14 +763,14 @@ namespace orc {
 
   std::unique_ptr<StripeStatistics>
   ReaderImpl::getStripeStatistics(uint64_t stripeIndex) const {
-    if (!isMetadataLoaded) {
+    if (!contents->metadata) {
```

Comment:
We may keep isMetadataLoaded flag to avoid repeated calls of readMetadata() if missing.

Reply:
+1 for @wgtmac 's comment keeping the existing `isMetadataLoaded` flag.

Reply:
Done